ckan/lib/search/index.py

```diff
@@ -245,6 +245,8 @@ def index_package(self, pkg_dict, defer_commit=False):
                         value = None
                 except ValueError:
                     continue
+                except TypeError:
```

Comment:
Stack trace:

2015-10-16 10:28:00,776 ERROR [ckan.lib.search] can't compare offset-naive and offset-aware datetimes
Traceback (most recent call last):
  File "/usr/lib/ckan/default/src/ckan/ckan/lib/search/__init__.py", line 103, in dispatch_by_operation
    index.update_dict(entity)
  File "/usr/lib/ckan/default/src/ckan/ckan/lib/search/index.py", line 103, in update_dict
    self.index_package(pkg_dict, defer_commit)
  File "/usr/lib/ckan/default/src/ckan/ckan/lib/search/index.py", line 240, in index_package
    if date != bogus_date:
TypeError: can't compare offset-naive and offset-aware datetimes

These lines:

if date != bogus_date:
                        value = date.isoformat() + 'Z'

Attempt to parse a date, and if the parse succeeds, converts it to isoformat and adds a Z. However, if the date is already in isoformat with a trailing Z it throws an exception. Perhaps instead of using bogus_date it may be wiser to simply check if the string is empty or None?